distributed/worker.py

```diff
@@ -3476,9 +3476,9 @@ def _deserialize(function=None, args=None, kwargs=None, task=no_value):
     """ Deserialize task inputs and regularize to func, args, kwargs """
     if function is not None:
         function = loads_function(function)
-    if args:
+    if args and isinstance(args, bytes):
         args = pickle.loads(args)
-    if kwargs:
+    if kwargs and isinstance(kwargs, bytes):
```

Comment:
Note that this is the only change that wasn't in this PR originally.  When constructing a task on the scheduler, we will need to have the option of using `to_serialize` on args/kwargs (so that we can avoid pickling `Serialized`/`Serialize` objects while still enabling function caching).  Therefore, these dict values may not be pickled bytes when we get to `_deserialize`.